src/control/client/config_test.go

```diff
@@ -0,0 +1,122 @@
+package client_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"reflect"
+	"testing"
+
+	"github.com/daos-stack/daos/src/control/client"
+	"github.com/daos-stack/daos/src/control/common"
+)
+
+func getDefaultConfig(t *testing.T) *client.Configuration {
+	t.Helper()
+
+	defaultConfig := client.NewConfiguration()
+	absPath, err := common.GetAbsInstallPath(defaultConfig.Path)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defaultConfig.Path = absPath
+
+	return defaultConfig
+}
+
+func getTestFile(t *testing.T) *os.File {
+	t.Helper()
+
+	testFile, err := ioutil.TempFile("", t.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return testFile
+}
+
+func TestLoadConfigDefaultsNoFile(t *testing.T) {
+	defaultConfig := getDefaultConfig(t)
+
+	cfg, err := client.GetConfig("")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !reflect.DeepEqual(cfg, defaultConfig) {
+		t.Fatalf("%v != %v", cfg, defaultConfig)
+	}
+}
+
+func TestLoadConfigFromDefaultFile(t *testing.T) {
+	defaultConfig := getDefaultConfig(t)
+	if err := os.MkdirAll(path.Dir(defaultConfig.Path), 0755); err != nil {
+		t.Fatal(err)
+	}
+	f, err := os.Create(defaultConfig.Path)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(defaultConfig.Path)
```

Comment:
why not just os.Remove?

Reply:
Hmm, good catch. I was thinking that it'd be the converse of MkdirAll, but of course for that to be true it should be working on a path, not the actual test file. Thinking about this some more, it's probably safest to just remove this line and trust that the whole test directory gets cleaned up.